File: algorithms/1_easy/M/02114_maximum_number_of_words_found_in_sentences/solution.cpp
class Solution {
public:
    int mostWordsFound(vector<string>& sentences) {
        int max{-1};

        for (const string &sentence: sentences) {
            int count{1};
            const size_t len{sentence.size()};

            for (size_t i{0}; i < len; ++i) {
                if (sentence[i] == ' ') {
                    count += 1;
                }
            }

            max = std::max(max, count);
        }

        return max;
    }
};

File: algorithms/1_easy/M/02114_maximum_number_of_words_found_in_sentences/solution.rs
impl Solution {
    pub fn most_words_found(sentences: Vec<String>) -> i32 {
        let mut max = 0;

        for sentence in sentences {
            let mut count = 1;
            for byte in sentence.into_bytes() {
                if byte == b' ' {
                    count += 1;
                }
            }

            max = max.max(count);
        }

        max
    }
}
